app/models/user.rb

```diff
@@ -318,4 +318,20 @@ def no_other_logins_exist?(login)
   def uploaded_subjects_count
     Subject.unscoped.where(upload_user_id: self.id).count
   end
+
+  def anonymize
+    self.email = "noreply-#{SecureRandom.hex(4)}@zooniverse.org"
+    self.current_sign_in_ip = nil
+    self.last_sign_in_ip = nil
+    self.display_name = self.id
```

Comment:
This might be visible publicly in some places, so I wonder if something like `"Deleted user #{self.id}"` would be better.